src/main/java/io/stargate/sgv2/jsonapi/service/resolver/model/impl/CreateCollectionCommandResolver.java

```diff
@@ -497,6 +498,18 @@ private Integer validateModelAndDimension(
       EmbeddingProvidersConfig.EmbeddingProviderConfig providerConfig,
       Integer userVectorDimension) {
     // Find the model configuration by matching the model name
+    // 1. huggingfaceDedicated does not require model, but requires dimension
+    if (userConfig.provider().equals(ProviderConstants.HUGGINGFACE_DEDICATED)) {
+      if (userVectorDimension == null) {
+        throw ErrorCode.INVALID_CREATE_COLLECTION_OPTIONS.toApiException(
```

Comment:
Use templating, `ProviderConstants.HUGGINGFACE_DEDICATED` for error.
Also quotes around `dimension`?

Reply:
Also, is it `'dimension'` or `'vectorDimension'`?

Reply:
thanks. fixed
yeah, user input is dimension